### PR TITLE
Implemented issue #403

### PR DESCRIPTION
**Pull request by mattdinthehouse**
Pretty much what the title says.

I didn't implement the ability to align what's selected purely because there wasn't any existing way to get the area selected. Instead it behaves the same way as the other transforms and does the operation on the whole canvas.

### CODE REVIEW

**Comment by juliandescottes** [comment]
Thanks for the PR!
I quickly looked at the code and it seems good. I will try to test and review in details later today.

No worries about the selection-related part, we'll handle this once we can easily access & modify the currently selected pixels.

File: src/js/tools/transform/TransformUtils.js [hunk 1]
```diff
@@ -62,6 +62,51 @@
...
       return frame;
+    },
+
+    align : function(frame) {
+      // Figure out the boundary
+      var minx = frame.width;
+      var miny = frame.height;
+      var maxx = 0;
+      var maxy = 0;
+      for (var col = 0; col < frame.width; col++) {
```
Comment (juliandescottes) [inline src/js/tools/transform/TransformUtils.js]: You could use frame.forEachPixel here to simplify the code.

File: src/js/tools/transform/TransformUtils.js [hunk 2]
```diff
@@ -62,6 +62,51 @@
       });
       frame.version++;
       return frame;
+    },
+
+    align : function(frame) {
```
Comment (juliandescottes) [inline src/js/tools/transform/TransformUtils.js]: Rename to "center" ?

File: src/js/tools/transform/Align.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+(function () {
+  var ns = $.namespace('pskl.tools.transform');
+
+  ns.Align = function () {
```
Comment (juliandescottes) [inline src/js/tools/transform/Align.js]: rename to Center?

**Comment by juliandescottes** [comment]
Optional: We could easily add a unit-test for this. You can have a look at TransformUtilsTest.js if you want to give it a shot. I think it should be pretty straightforward.

**Comment by juliandescottes** [comment]
Thanks, code change looks good and the feature works as expected. :+1:

Please have a look at my comments and let me know if you want to take care of them. Otherwise, I can log follow-ups and do it after merging your PR.

**Comment by mattdinthehouse** [comment]
Made those changes. "Align" is now "Center".

Also I couldn't run `grunt test` because karma wasn't able to load karma-phantomjs-launcher meaning it couldn't connect to PhantomJS. Hope you didn't mind me using Travis instead lol..

**Comment by juliandescottes** [comment]
Amazing work, thanks a lot! 
Sorry you had trouble with PhantomJS, I thought unit-tests could be run without any extra step. I'll have a look into that. 

Merging!

**Pull request merged**